### ISSUE TITLE
Use void argument list for generated Edger8r stubs

### ISSUE THREAD

**Issue by mikbras** [Microsoft/openenclave#971]
I encountered problems with generated Edger8r stubs for edge routines without any arguments in C. TheEdger8r generates edge routine prototypes like this:

```
int foo();
```

Then the enclave can define foo() with arbitrary arguments and the compiler does not detect the error because it treats the original definition as a K&R style definition.

Generating a prototype like this fixes the problem:

```
int foo(void);
```

**Comment by dthaler** [comment]
Correct. Similarly, Intel's sgx_edger8r.exe generates () and won't accept (void) in the EDL language.
But when you implement the ecall handler, you should use (void) in the implementation.

**Comment by anakrish** [comment]
Addressed in #976
